### DISCUSSION TITLE
Simple guide to install raspotify to Pi v1 and zero v1.x

### DISCUSSION THREAD

**Discussion by bluesmaz** [dtcooper/raspotify#543] [General]
Would there be any guide how to install raspotify 0.31.8.1 versio to headless PI v1 and zero v.1.x as they are not anymore officially supported? I really well understand why this is, and it is very much ok.  

I would assume that there is also others that are interested of this kind of alternative guide. :)

**Comment by JasonLG1979** [comment]
Think about what you're asking for, a guide to something that's unsupported. Right now the process is very similar to installing the current release but that could very well change at anytime.

**Reply by bluesmaz** [reply]
Yes. I understand this. 
Guide would be helpfull to understand the difference of official way with supported Pi´s and what needs to be done differently trying to set it up to unsupported Pi´s.

**Reply by JasonLG1979** [reply]
It's just like installing any other random .deb. download the .deb and run apt install ./random.deb.

**Reply by JasonLG1979** [reply]
Right now configuration is the same as far as I can remember.

**Reply by bluesmaz** [reply]
Excellent! This is helpfull.

**Reply by JasonLG1979** [reply]
Now that I'm home at my computer..

Specifically:

```
curl https://github.com/dtcooper/raspotify/releases/download/0.31.8.1/raspotify_0.31.8.1.librespot.v0.3.1-54-gf4be9bb_armhf.deb
```

```
sudo apt install ./raspotify_0.31.8.1.librespot.v0.3.1-54-gf4be9bb_armhf.deb
```

**Reply by JasonLG1979** [reply]
@bluesmaz I've added the instructions to the README.

**Comment by bluesmaz** [comment]
Excellent! Thank you a lot for this.

pe 20. toukok. 2022 klo 1.41 Jason Gray ***@***.***)
kirjoitti:

**Comment by Sshellcommando** [comment]
So I am stuck with a raspberry pi zero w until the prices come down. I used it as a wireless spotify connect point to play music from an older Home Theater receiver. I'd really like to get this working again somehow. I went to download the older version listed in this thread and couldn't get my raspberry pi to read a .deb file. I did some research and know the ARM6v is no longer supported by Debian (I think?) Would this old version of raspotify work if I flashed an older version of the Raspberry Pi operating system to the pi zero w? What can I do to get this working again?

**Reply by JasonLG1979** [reply]
I'm not sure what you mean by you couldn't get it to "read" a .deb?

You can use apt to install debs. `sudo apt install ./path/to/my/deb`

**Reply by JasonLG1979** [reply]
But to clarify on support, Debian does support ARMv6 in the form of `armel` `armhf` in vanilla Debian is ARMv7 +. The chip in the pi zero is sort of a half between. Raspberry Pi OS is patched to support the pi zero's chip. Vanilla `armhf` Debian does not support the pi zero.

**Reply by Sshellcommando** [reply]
I get this message
pi@raspberrypi:~ $ sudo apt install ./raspotify_0.31.8.1.librespot.v0.3.1-54-gf4be9bb_armhf.deb
Reading package lists... Done
E: Unsupported file ./raspotify_0.31.8.1.librespot.v0.3.1-54-gf4be9bb_armhf.deb given on commandline

**Reply by JasonLG1979** [reply]
What OS and what version of Debian is it based on?

**Reply by Sshellcommando** [reply]
rasbian 10 kernel 5.10.103+ not sure how to know the Dedian answer

**Reply by Sshellcommando** [reply]
Raspbian*

**Reply by JasonLG1979** [reply]
That would be why you'll need to use dpkg then. And you should probably update to Stable from Old Stable.

**Reply by Sshellcommando** [reply]
Ok. Thanks for the tip. Can you tell me why that will fix my problem? I have enjoyed learning all this stuff but still am a Noob.

**Reply by JasonLG1979** [reply]
Because the version of apt that comes Debian 10 doesn't support installing local debs.

**Reply by Sshellcommando** [reply]
Got this pi@raspberrypi:~ $ dpkg -install ./raspotify_0.31.8.1.librespot.v0.3.1-54-gf4be9bb_armhf.deb
dpkg: error: unknown option -n

Type dpkg --help for help about installing and deinstalling packages [*];
Use 'apt' or 'aptitude' for user-friendly package management;
Type dpkg -Dhelp for a list of dpkg debug flag values;
Type dpkg --force-help for a list of forcing options;
Type dpkg-deb --help for help about manipulating *.deb files;

Options marked [*] produce a lot of output - pipe it through 'less' or 'more' !

**Reply by JasonLG1979** [reply]
You should really just upgrade your OS.

**Reply by JasonLG1979** [reply]
I really am not trying to be mean but that package is offered as an unsupported out of date version for a device that is also not supported by me/Raspotify and this is starting to look a lot like support. Google and the help menu will tell you all you need to know, because that's basically what I'd be doing, Googling your questions.

**Reply by Sshellcommando** [reply]
That is fair and I certainly appreciate your help. I've put a lot of hours into trying to figure this out and googling this all myself and finally gave up to come here and just ask because I really don't understand what the issue is. Like I said I am mostly trying to learn.

**Reply by Sshellcommando** [reply]
WHen you say upgrade my os do you mean select a different OS or upgrade raspbian?

**Reply by JasonLG1979** [reply]
I mean install the latest stable Raspberry Pi OS. It's best to keep things as up to date as you can otherwise you'll end up with situations like this one where software passes you by.

**Reply by JasonLG1979** [reply]
To be clear 2 versions are currently offered. "Legacy" which is Old Stable in Debian speak and Stable. You want Stable. You're a version behind so you're running "Legacy" currently.

**Reply by vcenteary** [reply]
@Sshellcommando Im in Raspbian GNU/Linux 11 (bullseye), and had the same error as you when trying to install it with the prompts of Jason. The solution is using:

`wget https://github.com/dtcooper/raspotify/releases/download/0.31.8.1/raspotify_0.31.8.1.librespot.v0.3.1-54-gf4be9bb_armhf.deb`

and then

`sudo apt install ./raspotify_0.31.8.1.librespot.v0.3.1-54-gf4be9bb_armhf.deb`

Hope it works for you!

**Reply by Sshellcommando** [reply]
so does that allow the install of raspotify that works for the legacy 32 bit OS?

**Reply by vcenteary** [reply]
Very much so, I'm currently using raspotify in a Raspi 1 model b+, fully working and sounding great!

**Comment by victorclaessen** [comment]
For anyone stumbling onto this topic. The librespot version in the deb file mentioned above no longer works on my Pi 1/bookworm due to a missing library. I recompiled using cargo. This took forever, so in case anyone needs a working librespot binary, you can find it here. https://github.com/victorclaessen/librespot-pi1 . Hope that helps someone eventually. It's great that we can still use this old hardware in a useful way!

**Reply by PJBrs** [reply]
Hi, that sounds great! I assume this version will also work on the first raspberry pi zero? And what version of raspberry os? I think I'm still on bullseye...

**Reply by omarsayan** [reply]
Were you able to successfully run in on the zero? I can’t get it to work on the rpi zero w. I just put the librespot binary file from victorclaessen to /usr/bin. After which raspotify seemed to run and I got the device visible as a Spotify connect device in the iOS app. But when I try connect and play no sound comes out. Bluetooth speakers are connected to rpi

**Reply by victorclaessen** [reply]
At first glance that sounds maybe like a problem between librespot and your bluetooth speakers, so your ALSA configuration? Can you play anything else on your bluetooth speakers from the command line?

**Reply by omarsayan** [reply]
Thanks for replying! Yes I can play a simple wav file with “aplay -D file.wav”

**Reply by victorclaessen** [reply]
I, myself, have since moved on to the Home Assistant Snapcast-server Add-on, and now run only the snapclient on the raspberry Pi, so I'm having a hard time duplicating your setup.

I would suggest running the librespot binary just from the command line (you may need to look up the exact [command line options](https://github.com/librespot-org/librespot/wiki/Options) but it may look something like this `librespot --backend alsa --device my_bluetooth`.

You should then get the output of librespot in your terminal and may be able to use that to determine what's wrong when you play something. You can also add `--verbose` to get even more info.
Hope that helps.

If not, then look into snapserver/snapclient. I like it. Also gives you multiroom audio.

**Comment by victorclaessen** [comment]
I would imagine so. I ran it on (a fresh install of) the latest ~raspbian~ raspberry os bookworm.

**Comment by TheBobC** [comment]
Another “if you stumble across this” entry for a Pi Zero 2 w with 64 bit Bookworm lite. I simply wanted it to play default to a Bluetooth speaker. I worked for 2 months to get this working (sooo many different approaches out there). In the end, the working solution was maddeningly simple. 
1. Install Bookworm 64 bit lite (make sure ssh is enabled and wireless network info is entered)
2. sudo apt update
3. sudo apt upgrade
4.  sudo reboot
5.  sudo apt install bluez-alsa-utils 
6.  sudo nano /etc/asound.conf
7.  Copy paste this code:
          pcm.!default {
              type plug
              slave.pcm “bluealsa”
           }
           ctl.!default {
               type hw
               card “bluealsa”
            }
8. control x (to close nano)
9. Y (to confirm save)
10.  Enter
11. Now put the Bluetooth device (speaker) in discoverable mode, then on Pi
12. bluetoothctl 
13. scan on
14. wait to see your device
15.  note the MAC address (XX:XX:XX:XX:XX)
16.  scan off
17.  pair XX:XX:XX:XX:XX
18.  trust XX:XX:XX:XX:XX
19.  connect XX:XX:XX:XX:XX
20.  quit (to exit bluetoothctl)
21.  sudo apt-get -y install curl && curl -sL https://dtcooper.github.io/raspotify/install.sh | sh
22. sudo reboot
23.  turn off bluetooth device
24.  once Pi is restarted, turn on bluetooth device
25.  raspotify(pi name) should show up in Spotify as long as on same network 

This should be all you need. If Bluetooth does not pair initially, be certain the precursor before the MAC address is NEW when scanning. 

As you can see, there is no need for Pipewire nor Pulseaudio. (I spent hours upon hours going down those rabbit holes). Just updated/upgraded bookworm, bluez-alsa-utils and raspotify. Raspotify itself works fine without modification, although you can certainly do sudo nano /etc/raspotify/conf to change name, default stream quality etc. You should not enter username, backend, etc. 

Hope this helps someone. Good luck!

**Reply by TheBobC** [reply]
Indentations on the code above did not stick. I indented the first line 5 spaces and the other ones 7 spaces. Please see attached screenshot 


![Screenshot 2025-06-24 at 8 18 12 AM](https://github.com/user-attachments/assets/4def871a-7532-4548-b68f-27144ea9dfca)

**Comment by TheBobC** [comment]
This will also work on Raspberry Pi 5. Same process with a slight change to the asound.conf:

pcm.btreceiver { 
    type plug 
    slave.pcm { 
        type bluealsa 
        device "8B:13:69:8F:42:75"
        profile "a2dp" 
    } 
    hint { 
        show on 
        description "Bluetooth Receiver" 
    } 
}

pcm.!default {
    type plug
    slave.pcm "btreceiver"
}

**Comment by TheBobC** [comment]
Same issue with not showing indentations...
<img width="392" alt="Screenshot 2025-07-06 at 10 06 31 AM" src="https://github.com/user-attachments/assets/94e9863d-4d69-463f-84a2-ac960415a0fc" />

**Comment by kimtore** [comment]
@TheBobC for indentation, please wrap the code in triple backticks (```)

```
pcm.btreceiver {
    type plug
    slave.pcm {
        type bluealsa
        device "8B:13:69:8F:42:75"
        profile "a2dp"
    }
    hint {
        show on
        description "Bluetooth Receiver"
    }
}

pcm.!default {
    type plug
    slave.pcm "btreceiver"
}
```
